kotlin/lib/src/main/kotlin/goodmetrics/Metrics.kt

```diff
@@ -1,5 +1,22 @@
 package goodmetrics
 
+sealed interface SupportedDimensionType
+data class StringDimension(val value: String) : SupportedDimensionType {

```

Comment:
How about
```
sealed interface Dimension {
  val name: String

  data class String(val name: String, val value: String) : Dimension
  data class Number(val name: String, val value: Long) : Dimension
  data class B[...]
}
```

which would show up in use a little nicer as `Dimension.String` and whatnot, while things that only care about the interface get to take a `Dimension`.

Could go here or nested inside of `Metrics` to make it `Metrics.Dimension.String` if you prefer

Reply:
Works for me, happy to nest it inside of `Metrics`